### ISSUE TITLE
INatIconButton passess accessibility tests even when it's not accessible

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistReactNative#745]
**Describe the bug**
The following will pass even though the button is just an icon with nothing useful for a screen reader to read:

```javascript
expect(
  <INatIconButton icon="camera" onPress={jest.fn( )} />
).toBeAccessible( );
```

**To Reproduce**
1. Make a test like above and watch it pass

**Expected behavior**
A button with no `accessibilityLabel` should not pass an accessibility test.

**Additional context**
I suspect this happens because `toBeAccessible` looks at `INatIconButton`, sees it has text content in the form of a letter in our icon font, and assumes that's good enough for a screen reader, but for us it's not, because that text will make no sense to anyone.

IMO, we need to make a few changes that will create a lot of refactoring work:

- [ ] `INatIconButton` should require the `accessibilityLabel` prop and throw an exception if it doesn't get it or gets a blank value (null, undefined, zero-length string, etc)
- [ ] If screen readers are reading the icon font text, stop them from doing so (they should read the `accessibilityLabel` instead)
